local-storage/src/provider.rs

```diff
@@ -166,20 +221,29 @@ impl StorageProvider for SqliteStorageProvider {
         }
     }
 
-    fn list_available_dags(&self) -> Result<Vec<String>> {
+    fn list_available_dags(&self) -> Result<Vec<(String, String)>> {
         let roots = self
             .conn
-            .prepare("SELECT DISTINCT root_cid FROM links")?
+            .prepare("SELECT DISTINCT cid, filename FROM blocks")?
             .query_map([], |row| {
                 let cid_str: String = row.get(0)?;
-                Ok(cid_str)
+                let filename_str: String = row.get(1)?;
+                Ok((cid_str, filename_str))
             })?
             // TODO: Correctly catch/log/handle errors here
             .filter_map(|cid| cid.ok())
             .collect();
         Ok(roots)
     }
 
+    fn name_dag(&self, cid: &str, file_name: &str) -> Result<()> {
```

Comment:
Conversely this method assumes you want to drop the _old_ name.

I would've guessed there'd be a parameter, perhaps bool or bool-equivalent-enum, specifying whether you'd like to do this or do `WHERE cid = ?2 AND filename = NULL`